File: samples/snippets/csharp/VS_Snippets_VBCSharp/VbRaddataDisplaying/CS/LookupBox.cs
//<Snippet5>
using System.Windows.Forms;

namespace CS
{
    [System.ComponentModel.LookupBindingProperties("DataSource", "DisplayMember", "ValueMember", "LookupMember")]
    public partial class LookupBox : UserControl
    {
        public object DataSource
        {
            get{ return comboBox1.DataSource; }
            set{ comboBox1.DataSource = value; }
        }

        public string DisplayMember
        {
            get{ return comboBox1.DisplayMember; }
            set{ comboBox1.DisplayMember = value; }
        }

        public string ValueMember
        {
            get{ return comboBox1.ValueMember; }
            set{ comboBox1.ValueMember = value; }
        }

        public string LookupMember
        {
            get{ return comboBox1.SelectedValue.ToString(); }
            set{ comboBox1.SelectedValue = value; }
        }


        public LookupBox()
        {
            InitializeComponent();
        }
    }
}
//</Snippet5>

File: samples/snippets/visualbasic/VS_Snippets_VBCSharp/VbRaddataDisplaying/VB/LookupBox.vb
'<Snippet5>
<System.ComponentModel.LookupBindingProperties("DataSource", "DisplayMember", "ValueMember", "LookupMember")>
Public Class LookupBox

    Public Property DataSource() As Object
        Get
            Return ComboBox1.DataSource
        End Get
        Set(ByVal value As Object)
            ComboBox1.DataSource = value
        End Set
    End Property

    Public Property DisplayMember() As String
        Get
            Return ComboBox1.DisplayMember
        End Get
        Set(ByVal value As String)
            ComboBox1.DisplayMember = value
        End Set
    End Property

    Public Property ValueMember() As String
        Get
            Return ComboBox1.ValueMember
        End Get
        Set(ByVal value As String)
            ComboBox1.ValueMember = value
        End Set
    End Property

    Public Property LookupMember() As String
        Get
            Return ComboBox1.SelectedValue.ToString()
        End Get
        Set(ByVal value As String)
            ComboBox1.SelectedValue = value
        End Set
    End Property
End Class
'</Snippet5>
